test/parallel/test-querystring-maxKeys-Infinity.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+// This test was originally written to test a regression
+// that was introduced by
+// https://github.com/nodejs/node/pull/2288#issuecomment-179543894
+require('../common');
+
+const assert = require('assert');
+const parse = require('querystring').parse;
+
+// taken from express-js/body-parser
+function createManyParams(count) {
+  var str = '';
+
+  if (count === 0) {
+    return str;
+  }
+
+  str += '0=0';
+
+  for (var i = 1; i < count; i++) {
+    var n = i.toString(36);
+    str += '&' + n + '=' + n;
+  }
+
+  return str;
+}
+
+const count = 10000;
+const params = createManyParams(count);
+
+// thealphanerd
+// 27def4f introduced a change to parse that would cause Inifity
+// to be passed to String.prototype.split as an argument for limit
+// In this instance split will always return an empty array
+// this test confirms that the output of parse is the expe4cted length
+// when passed Infinity as the argument for maxKeys
+const result = parse(params, undefined, undefined, {maxKeys: Infinity});
```

Comment:
@thefourtheye I've updated the test to include tests for all four "non finite" values